freqtrade/leverage/liquidation_price.py

```diff
@@ -0,0 +1,351 @@
+from typing import Optional
+
+from freqtrade.enums import Collateral, TradingMode
+from freqtrade.exceptions import OperationalException
+
+
+def liquidation_price(
+    exchange_name: str,
+    open_rate: float,
+    is_short: bool,
+    leverage: float,
+    trading_mode: TradingMode,
+    collateral: Optional[Collateral],
+    # Binance
+    wallet_balance: Optional[float] = None,
+    mm_ex_1: Optional[float] = None,
+    upnl_ex_1: Optional[float] = None,
+    maintenance_amt: Optional[float] = None,
+    position: Optional[float] = None,
+    mm_rate: Optional[float] = None,
+    # Gateio
+    collateral_amount: Optional[float] = None,
+    contract_size: Optional[float] = None,
+    num_contracts: Optional[float] = None,
+    taker_fee: Optional[float] = None,
+    # TODO-lev: Check that gatio margin ratio and okex margin ratio are the same
+    mm_ratio: Optional[float] = None,
+    # Okex
+    liability: Optional[float] = None,
+    interest: Optional[float] = None,
+    taker_fee_rate: Optional[float] = None,  # TODO-lev: Check if same as gateio taker_fee
+    position_assets: Optional[float] = None,
+) -> Optional[float]:
```

Comment:
@frosty00 Do you know how many of these variables are actually the same thing? It looks like each exchange gave a different formula for how they calculate liquidation prices, but ideally it'd be better to just have 1 formula for all exchanges (I'm only looking at isolated liquidation price to begin with)

Reply:
@samgermain take a look at my edits to kucoinfutures and gateio `fetchPositions`

Reply:
it looks like at least collateral and collateral amount are the same, i'd advise to use the same naming for position variables as we do in ccxt

Reply:
> i'd advise to use the same naming for position variables as we do in ccxt

can't agree more on this :+1:

Reply:
> it looks like at least collateral and collateral amount are the same, i'd advise to use the same naming for position variables as we do in ccxt

They're actually not, `collateral` should probably be renamed to margin_mode or something, it just means either `isolated` or `cross`

> i'd advise to use the same naming for position variables as we do in ccxt

If I knew how to do that I don't think I would need any help here. The problem is that I don't know exactly what all these things are, and I'm trying to figure out which ones are equivelent. 

I'm also having trouble finding the maintenance margin rate and maintenance amount from ccxt. This is for dry run, so I won't actually be able to call the `fetchPositions` method because there won't actually be any positions on the exchange

Reply:
the maintenance margin rate and maintenance amount normally depend on the size of the position

Reply:
> They're actually not, collateral should probably be renamed to margin_mode or something, it just means either isolated or cross

we call this `marginType` in ccxt

Reply:
> the maintenance margin rate and maintenance amount normally depend on the size of the position

How so. How can I get the maintenance margin rate and maintenance amount from the position size?